From bbf593fe89706a7cbebc165f737c26f1358d663b Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Thu, 30 Jul 2015 16:35:52 -0500
Subject: [PATCH 0744/5242] MLK-11335: mtd: qspi: enalbe LPSR for qspi

commit  35f3b130de9215096d29c7962cd1ff7f713230ae from
https://source.codeaurora.org/external/imx/linux-imx.git

The LPSR turns off the power for IOMUX when suspending so restore the
IOMUX when resuming.

Signed-off-by: Han Xu <b45815@freescale.com>
(cherry picked from commit 906d0c1381e865dc7c96a4bde6fe694f1ce089a9)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index 3d0e480..3d7bf33 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -1194,6 +1194,7 @@ static int fsl_qspi_remove(struct platform_device *pdev)
 
 static int fsl_qspi_suspend(struct platform_device *pdev, pm_message_t state)
 {
+	pinctrl_pm_select_sleep_state(&pdev->dev);
 	return 0;
 }
 
@@ -1202,6 +1203,8 @@ static int fsl_qspi_resume(struct platform_device *pdev)
 	int ret;
 	struct fsl_qspi *q = platform_get_drvdata(pdev);
 
+	pinctrl_pm_select_default_state(&pdev->dev);
+
 	ret = fsl_qspi_clk_prep_enable(q);
 	if (ret)
 		return ret;
-- 
1.7.9.5

